op-acceptance-tests/tests/jovian/da_footprint_test.go

```diff
@@ -34,62 +42,190 @@ func (s *CalldataSpammer) Spam(t devtest.T) error {
 	return err
 }
 
+type daFootprintSystemConfig struct {
+	SetDAFootprintGasScalar func(scalar uint16) bindings.TypedCall[any] `sol:"setDAFootprintGasScalar"`
+	DAFootprintGasScalar    func() bindings.TypedCall[uint16]           `sol:"daFootprintGasScalar"`
+}
+
+type daFootprintL1Block struct {
+	DAFootprintGasScalar func() bindings.TypedCall[uint16] `sol:"daFootprintGasScalar"`
+}
+
+type daFootprintEnv struct {
+	l1Client     *dsl.L1ELNode
+	l2Network    *dsl.L2Network
+	l2EL         *dsl.L2ELNode
+	systemConfig daFootprintSystemConfig
+	l1Block      daFootprintL1Block
+}
+
+func newDAFootprintEnv(t devtest.T, l2Network *dsl.L2Network, l1EL *dsl.L1ELNode, l2EL *dsl.L2ELNode) *daFootprintEnv {
+	systemConfig := bindings.NewBindings[daFootprintSystemConfig](
+		bindings.WithClient(l1EL.EthClient()),
+		bindings.WithTo(l2Network.Escape().Deployment().SystemConfigProxyAddr()),
+		bindings.WithTest(t),
+	)
+
+	l1Block := bindings.NewBindings[daFootprintL1Block](
+		bindings.WithClient(l2EL.Escape().EthClient()),
+		bindings.WithTo(common.HexToAddress("0x4200000000000000000000000000000000000015")),
+		bindings.WithTest(t),
+	)
+
+	return &daFootprintEnv{
+		l1Client:     l1EL,
+		l2Network:    l2Network,
+		l2EL:         l2EL,
+		systemConfig: systemConfig,
+		l1Block:      l1Block,
+	}
+}
+
+func (env *daFootprintEnv) checkCompatibility(t devtest.T) {
+	// Ensure getters exist on both L1 SystemConfig and L2 L1Block
+	_, err := contractio.Read(env.systemConfig.DAFootprintGasScalar(), t.Ctx())
+	t.Require().NoError(err)
+	_, err = contractio.Read(env.l1Block.DAFootprintGasScalar(), t.Ctx())
+	t.Require().NoError(err)
+}
+
+func (env *daFootprintEnv) getSystemConfigOwner(t devtest.T) *dsl.EOA {
+	priv := env.l2Network.Escape().Keys().Secret(devkeys.SystemConfigOwner.Key(env.l2Network.ChainID().ToBig()))
+	return dsl.NewKey(t, priv).User(env.l1Client)
+}
+
+func (env *daFootprintEnv) setDAFootprintScalarViaSystemConfigOnL1(t devtest.T, scalar uint16) *types.Receipt {
+	owner := env.getSystemConfigOwner(t)
+	rec, err := contractio.Write(env.systemConfig.SetDAFootprintGasScalar(scalar), t.Ctx(), owner.Plan())
+	t.Require().NoError(err, "SetDAFootprintGasScalar transaction failed")
+	t.Logf("Set DA footprint gas scalar on L1: scalar=%d", scalar)
+	return rec
+}
+
+// expectDAFootprintScalar expects the given DA footprint gas scalar to be set in the L1Block contract.
+func (env *daFootprintEnv) expectDAFootprintScalar(t devtest.T, expected uint16) {
+	current, err := contractio.Read(env.l1Block.DAFootprintGasScalar(), t.Ctx())
+	t.Require().NoError(err, "Failed to read DA footprint gas scalar from L1Block")
+	t.Require().Equal(expected, current)
+}
```

Comment:
```suggestion
// expectDAFootprintScalar expects the given DA footprint gas scalar to be set in the L1Block contract.
func (env *daFootprintEnv) expectDAFootprintScalarInL1BlockPredeploy(t devtest.T, expected uint16) {
	current, err := contractio.Read(env.l1Block.DAFootprintGasScalar(), t.Ctx())
	t.Require().NoError(err, "Failed to read DA footprint gas scalar from L1Block")
	t.Require().Equal(expected, current)
}
```
(requires changes to the calling code, too, obvs).